modules/scratchFetch.ts

```diff
@@ -0,0 +1,218 @@
+import { defineEvent } from "strife.js";
+import fetch from "node-fetch";
+import { getSettings } from "./settings.js";
+
+defineEvent("messageCreate", async (message) => {
+	const notSet = getSettings(message.author, false)?.scratchEmbeds === undefined;
+	if (!getSettings(message.author, false).scratchEmbeds && !notSet) {
+		return;
+	}
+	const scratchUrlRegex =
+		/(?<!<)https?:\/\/scratch\.mit\.edu\/(projects|users|studios)\/\w+\/?(?!>)/; //gpt wrote the regex and like half of this code
+	const match = message.content.match(scratchUrlRegex);
+
+	if (!match) {
+		return;
+	}
+
+	const urlParts = match[0].split("/");
+	const type = urlParts[3]; // m ybrain is dead
+
+	async function fetchApiData(apiUrl: string): Promise<any> {
+		try {
+			const response = await fetch(apiUrl);
+
+			const data = await response.json();
+			return data;
+		} catch (error) {}
+	}
+	function long(text: string, maxLength: number, appendText: string): string {
+		if (text.length > maxLength) {
+			return text.substr(0, maxLength) + appendText;
+		}
+		return text;
+	}
+
+	if (type === "projects") {
+		const projectId = urlParts[4];
+
+		const apiUrl = `https://api.scratch.mit.edu/projects/${projectId}/`;
+
+		fetchApiData(apiUrl)
+			.then((data) => {
+				//I hate hate hate hate this aaaaa
+
+				message.channel.send({
+					embeds: [
+						{
+							title: data.title,
+							description: `**desc**: ${data.description}\n**inst**: ${data.instructions}`,
```

Comment:
use non-inline fields
also fix your capitaliztion lmao

Reply:
> fix your capitaliztion lmao
i was planning to use emojis instead of text, the text was a placeholder